core/src/Kokkos_HIP.hpp

```diff
@@ -131,8 +131,8 @@ class HIP {
    */
   struct SelectDevice {
     int hip_device_id;
-    SelectDevice() : hip_device_id(1) {}
-    explicit SelectDevice(int id) : hip_device_id(id + 1) {}
+    SelectDevice() : hip_device_id(0) {}
+    explicit SelectDevice(int id) : hip_device_id(id) {}
```

Comment:
Indexing starts at 0?

Reply:
yeah so it's kind of strange. If you do `nvidia-smi` you will see the first gpu is zero but if you do `rocm-smi` the first gpu is one. However, when you get the device id, it is zero.